cfme/test_framework/sprout/plugin.py

```diff
@@ -255,3 +256,23 @@ def ping_pool(self):
             timeout = 60 + random.randint(0, 10)
         finally:
             self.reset_timer(timeout=timeout)
+
+
+def pytest_addhooks(pluginmanager):
+    pluginmanager.add_hookspecs(NewHooks)
+
+
+def pytest_miq_node_shutdown(config, nodeinfo):
+    ip_address = urlparse(nodeinfo).netloc
```

Comment:
Just one note here - I don't know what "nodeinfo" url looks like, but if it contains port than (e.g. "http://1.2.3.4:80/something") then ip_address will be "1.2.3.4:80" - the port will be included. Just FYI.